src/mqt/bench/error_correction/shor_transpiler.py

```
@@ -0,0 +1,431 @@
+                self.bit_flip_syndrome,
+                self.phase_flip_syndrome,
+                self.bit_flip_measure,
+                self.phase_flip_measure,
+            ])
+        return regs
+
+
+class ShorTranspiler:
+    """A high-level transpiler that encodes a QuantumCircuit using Shor's 9-qubit error correction code."""
+
+    def __init__(self, original_circuit: QuantumCircuit, add_syndromes: bool = True) -> None:
```

Ruff (0.15.18)

[warning] Boolean-typed positional argument in function definition

(FBT001)